Scenarios/StepArguments.swift

```diff
@@ -1,20 +1,24 @@
 //  Copyright © 2015 Outware Mobile. All rights reserved.
 
-public struct StepArguments: Indexable {
+public struct StepArguments: Collection {
 
   public let startIndex: Int = 0
   public var endIndex: Int { return captures.endIndex }
 
-  private let captures: [String]
+  fileprivate let captures: [String]
 
   internal init(_ matchResult: MatchResult) {
-    self.captures = matchResult.captures
+    self.captures = matchResult.captures.flatMap { $0 }
   }
 
   public subscript(index: Int) -> String {
     return captures[index]
   }
 
+  public func index(after i: Int) -> Int {
+    return i + 1
```

Comment:
Yeah, if the implementation isn't meant to be explicitly `i + 1`, I think that would be more representative of the type.